modules/azure/disk.go

```diff
@@ -0,0 +1,101 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// DiskExists indicates whether the specified Azure Managed Disk exists
+// This function would fail the test if there is an error.
+func DiskExists(t testing.TestingT, diskName string, resGroupName string, subscriptionID string) bool {
+	exists, err := DiskExistsE(diskName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// DiskExistsE indicates whether the specified Azure Managed Disk exists
+func DiskExistsE(diskName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get the Disk object
+	_, err := GetDiskE(diskName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetDiskType returns the Disk Storage Account Type of the Azure Managed Disk
+// This function would fail the test if there is an error.
+func GetDiskType(t testing.TestingT, diskName string, resGroupName string, subscriptionID string) compute.DiskStorageAccountTypes {
```

Comment:
I can see having a `DiskExists` method, as it adds some value over the underlying `GetDisk` (the check for `ResourceNotFoundErrorExists`), but `GetDiskType` is just a very thin wrapper over `GetDisk`... Is it really worth adding/maintaining?

Reply:
@brikis98 normally I would agree but we found it is a very common operation in test to identify the OS disk or data disk to perform some additional testing against.